src/orion/core/worker/consumer.py

```diff
@@ -30,7 +30,16 @@
 class ExecutionError(Exception):
     """Error raised when Orion is unable to execute the user's script without errors."""
 
-    pass
+    def __init__(self, return_code=0):
+        super(ExecutionError, self).__init__()
+        self.return_code = return_code
+
+
+def _exec_error(return_code):
```

Comment:
A simple test involving pickling the error returned by execute_process would help ensure that no future changes incompatible with pickling are introduced.

Reply:
what

Reply:
Adding a test where we call Consumer.execute_process and verify that the output can be pickled will cover this part and ensure we do not introduce unnoticed changes that make the output of Consumer.execute_process unpicklable.

Reply:
Alright, fixed here: https://github.com/Epistimio/orion/pull/684/commits/cefca8bfdd713ff62584d6bb8cd60b7fcebca7cc